docker-app/worker_wrapper/wrapper.py

```diff
@@ -286,6 +287,22 @@ def after_docker_run(self) -> None:
             self.job.project.last_package_job = self.job
             self.job.project.save()
 
+            try:
+                project_id = str(self.job.project.id)
+                package_ids = storage.get_stored_package_ids(project_id)
+
+                for package_id in package_ids:
+                    # keep the last package
+                    if package_id == str(self.job.project.last_package_job_id):
+                        continue
+
+                    storage.delete_stored_package(project_id, package_id)
+            except Exception as err:
+                logger.error(
+                    "Failed to delete dangling packages, will be deleted via CRON later.",
+                    exc_info=err,
+                )
```

Comment:
As said above, maybe better not to delete them too soon ?

And anyway, IMO better to use just one of these two implementations (either cron OR after successful package job), unless there is a particular reason we expect it to sometimes fail on the job ?

Reply:
I always assume the job might crash, so it would be good if we have a self cleaning mechanism. Why do you prefer having only one of the cleanup mechanisms?

Reply:
Addressed with the new check for the job id.

Reply:
re: two places same logic
I prefer to keep it like this, so the storage is available ASAP for the users.